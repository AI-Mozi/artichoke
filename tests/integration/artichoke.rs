use insta;
use crate::run;
use std::fs;
use std::path::Path;

#[test]
fn test_ruby_apps() -> Result<(), String> {
    let binary_name = "artichoke";
    let app_paths = Path::new("./tests/integration/apps");

    let paths = fs::read_dir(app_paths).unwrap();

    for path in paths {
        let file_path = path.unwrap().path();

        let str_path = file_path.clone().into_os_string().into_string().unwrap();
        let args = vec![&*str_path];

        let snapshot_name = format!("ruby_app__{}", file_path.strip_prefix(app_paths).unwrap().display());
        insta::assert_debug_snapshot!(snapshot_name, run(binary_name, args))
    }

    Ok(())
}
