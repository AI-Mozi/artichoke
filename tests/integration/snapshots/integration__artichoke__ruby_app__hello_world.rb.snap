---
source: tests/integration/artichoke.rs
assertion_line: 20
expression: "run(binary_name, args)"

---
Ok(
    CommandOutput {
        call_args: [
            "./tests/integration/apps/hello_world.rb",
        ],
        status: 0,
        stdout: "Hello Artichoke!\n",
        stderr: "",
    },
)
